### PR TITLE
Client type configuration inheritance

### LINKED ISSUE #30213: Client types configuration inheritance
Implementation for Client Type inheritance.

Can specify "parent" client type field to inherit defined client type fields of a parent.

### PR DESCRIPTION
**Pull request by patrickjennings**
Implementation for Client Type inheritance.

Can specify "parent" client type field to inherit defined client type fields of a parent.

closes #30213

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@patrickjennings Thanks for the PR!

I am approving, but did not test it and not 100% sure about various scenarios (like for instance if inheritance of 3 levels or more is supported as well etc, but even if it is not, I would go with the PR and possibly address other things as a follow-up if needed).

@vickeybrown @jsorah Does the PR look ok to you?

**Comment by mposolda** [comment]
FYI. created issue #30213 as it seems there was no GH issue associated with this.

File: services/src/main/java/org/keycloak/services/clienttype/impl/DefaultClientType.java [hunk 1]
```diff
@@ -44,31 +47,40 @@ public String getName() {
...
+        ClientTypeRepresentation.PropertyConfig propertyConfig = clientType.getConfig().get(optionName);
+        if (propertyConfig != null) {
+            return propertyConfig.getApplicable();
+        }
+
+        if (parentClientType != null) {
+            return parentClientType.isApplicable(optionName);
+        }
+
+        return true;
```
[the remark this answers is not in the document]
Comment (mposolda) [reply services/src/main/java/org/keycloak/services/clienttype/impl/DefaultClientType.java:59]: Added https://github.com/keycloak/keycloak/issues/30214

**Review by vickeybrown — APPROVED** [review]

**Comment by mposolda** [comment]
@vickeybrown Thanks for the review!

@patrickjennings I was not able to merge this because of conflicts. Could you please rebase this? Hopefully rebase will help to solve the conflicts.

**Comment by patrickjennings** [comment]
@mposolda rebased the branch and resolved the conflicts.

**Review by mposolda — APPROVED** [review]
@patrickjennings Thanks!

**Pull request merged**